### PR TITLE
Expose S3 http api

### PR DESCRIPTION
**Pull request by chunningham**
Provides a HTTP API for the KV data store at `/<orbit_id>/s3/<key?>`. Also:
- generalizes `Action` to use `String` instead of `Cid`
- use existing auth guards for the S3 api
- extract `OrbitMetadata` generation from `create_orbit`
- expand auth checks for orbit creation
- resurrect `/integration_tests`

### CODE REVIEW

**Review by sbihel — APPROVED** [review]
I haven't read everything in small detail but I went through the whole thing. I think some things could be polished but I don't really have anything to offer at the moment. (this is already good)

File: src/ipfs.rs [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
     raw::RawCodec,
-    store::DefaultParams,
+    store::StoreParams,
+    IpldCodec,
 };
 
+#[derive(Clone, Debug, Default)]
+pub struct KeplerParams;
+
+impl StoreParams for KeplerParams {
```
[the remark this answers is not in the document]
Comment (chunningham) [reply src/ipfs.rs]: its a conventient place to change block size if we need to, not strictly necessary now

File: src/s3_routes.rs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+            .collect::<Result<Vec<String>>>()
+            .map_err(|e| (Status::InternalServerError, e.to_string()))?,
+    ))
+}
+
+#[head("/<_orbit_id>/s3/<key..>")]
+pub async fn get_metadata(
+    _orbit_id: CidWrap,
+    orbit: GetAuthWrapper,
+    key: PathBuf,
```
[the remark this answers is not in the document]
Comment (chunningham) [reply src/s3_routes.rs:119]: it implements `FromSegments`, while `String` does not. I found it the easiest way to extract the remaining path.

File: src/orbit.rs [hunk 3]
```diff
@@ -93,33 +130,99 @@ impl AuthorizationToken for AuthTokens {
...
+}
+
+pub async fn get_metadata(
+    oid: &Cid,
+    param_str: &str,
+    chains: &ExternalApis,
+) -> Result<OrbitMetadata> {
+    let (method, params) = verify_oid(oid, param_str)?;
+    Ok(match (method.as_str(), &chains) {
+        ("tz", ExternalApis { tzkt, .. }) => params_to_tz_orbit(*oid, &params, &tzkt).await?,
```
[the remark this answers is not in the document]
Comment (chunningham) [reply src/orbit.rs:185]: The metadata type represents the manifest basically 1:1, perhaps it should be renamed for clarity

**Review by sbihel — APPROVED** [review]

File: src/routes.rs [hunk 4]
```diff
@@ -183,14 +182,10 @@ pub async fn delete_content(
         .map_err(|_| (Status::InternalServerError, "Failed to delete content"))?)
 }
 
-#[post("/<_orbit_id>", format = "text/plain", data = "<_params_str>")]
-pub async fn open_orbit_authz(
-    _orbit_id: CidWrap,
-    _params_str: &str,
-    _authz: CreateAuthWrapper,
-) -> Result<(), (Status, &'static str)> {
+#[post("/")]
```
[the remark this answers is not in the document]
Comment (chunningham) [reply src/routes.rs]: restored

**Pull request merged**